main.py

```diff
@@ -204,34 +181,161 @@ def create_circuit(self):
-                 "second": 00
-           }
+    # METHODS TO HANDLE STOREHOUSE
+    def create_box(self):
+        """Create a new box."""
+        content = {'namespace': self.namespace,
+                   'callback': self._create_box_callback,
+                   'data': {}}
+        event = KytosEvent(name='kytos.storehouse.create', content=content)
+        self.controller.buffers.app.put(event)
+        log.info('Create box from storehouse.')
+
+    def _create_box_callback(self, event, data, error):
+        """Callback to handle create_box."""
+        if error:
+            log.error(f'Can\'t create box with namespace {self.namespace}')
+
+        self.box = data
+        log.info(f'Box {self.box.box_id} was created in {self.namespace}.')
+
+    def list_stored_boxes(self):
+        """List all boxes using the current namespace."""
+        name = 'kytos.storehouse.list'
+        content = {'namespace': self.namespace,
+                   'callback': self._handle_boxes}
+
+        event = KytosEvent(name=name, content=content)
+        self.controller.buffers.app.put(event)
+        log.info(f'Bootstraping storehouse box for {self.namespace}.')
+
+    def _handle_boxes(self, event, data, error):
+        """Create a new box or retrieve a stored box."""
+        if len(data) == 0:
+            self.create_box()
+        else:
+            self.get_stored_box(data[0])
+
+    def get_stored_box(self, box_id):
+        """Get box from storehouse"""
+        content = {'namespace': self.namespace,
+                   'callback': self._get_box_callback,
+                   'box_id': box_id,
+                   'data': {}}
+        name = 'kytos.storehouse.retrieve'
+        event = KytosEvent(name=name, content=content)
+        self.controller.buffers.app.put(event)
+        log.info(f'Retrieve box with {box_id} from {self.namespace}.')
+
+    def _get_box_callback(self, event, data, error):
+        """Handle get_box method saving the box or logging with the error."""
+        if error:
+            log.error(f'Box {data.box_id} not found in {data.namespace}.')
+
+        self.box = data
+        log.info(f'Box {self.box.box_id} was load from storehouse.')
+
+    def save_circuit(self, circuit):
+        """Save a circuit using the storehouse."""
+        self.box.data[circuit.id] = circuit.as_dict()
+
+        content = {'namespace': self.namespace,
+                   'box_id': self.box.box_id,
+                   'data': self.box.data,
+                   'callback': self._save_circuit_callback}
+
+        event = KytosEvent(name='kytos.storehouse.update', content=content)
+        self.controller.buffers.app.put(event)
+
+    def _save_circuit_callback(self, event, data, error):
+        """Callback to handle save circuit."""
+        if error:
+            log.error(f'Can\'t update the {self.box.box_id}')
+
+        log.info(f'Box {data.box_id} was updated.')
+
+    @listen_to('.*kytos/of_core.handshake.completed')
```

Comment:
Here, we should not listen to of_core events. Remember: we are creating a NApp that the future cannot not use openflow anymore. So, please listen to topology "agnostic" events.